src/db_fns.py

```diff
@@ -66,7 +66,61 @@ def insert_film_db(engine,
         cnx.execute(query_full, details2)
     
     return
+    
+#%% Functions to get rated films
+def get_user_rated_films(engine, user_name):
+    # Connect to database - this works but outputs an sql error for some reason
+    engine.connect()
+    # Run query for table
+    df_films = pd.read_sql_query("""
+    SELECT b.film_key, b.title, b.year, a.rating
+    FROM W2W_User_Rating a 
+    INNER JOIN W2W_Films b 
+    on a.film_key = b.film_key 
+    where a.user_key = %s
+    order by timestamp_ur desc""", engine, params=(hashlib.md5(user_name.encode()).hexdigest(),))
+    return df_films
 
+def get_group_rated_films(engine, group_name):
+    
+    # Connect to database - this works but outputs an sql error for some reason
+    engine.connect()
+    # Run query for table
+    df_films = pd.read_sql_query("""
+    SELECT c.film_key, c.title, c.year, avg(rating) as group_rating, count(rating) as number_of_ratings
+    FROM W2W_User_Rating a 
+    LEFT JOIN W2W_Group_User_Mapping b 
+    on a.user_key = b.user_key 
+    INNER JOIN W2W_Films c
+    on a.film_key=c.film_key
+    where b.group_key = %s
+    group by c.title, c.year
+    order by group_rating desc
+    """, engine, params=(hashlib.md5(group_name.encode()).hexdigest(),))
+    return df_films
```

Comment:
In the streamlit app, we already get and cache the complete film table.
This could be adapted to use that dataframe rather than require a new extract from the database.